File: Quorum/Library/Standard/Plugins/javascript/Libraries/Language/Types/Boolean.js
class plugins_quorum_Libraries_Language_Types_Boolean_ {
    constructor(optional) {
        this.myBoolean = false;
    
        if(optional === undefined) {
        } else {
            this.myBoolean = optional;
        }
    }
    
    SetValueNative$quorum_boolean(bool) {
        myBoolean = bool;
    };
    
    GetHashCode() {
        return this.myBoolean ? 1231 : 1237;;
    };
}

Boolean.prototype.GetText = function() {
    return this.valueOf() + "";
};

Boolean.prototype.GetValue = function() {
    return this.valueOf().myBoolean;
};

Boolean.prototype.Equals$quorum_Libraries_Language_Object = function (value) {
    return this.valueOf().myBoolean === value;
};

Boolean.prototype.Compare$quorum_Libraries_Language_Object = function(value) {
    var b = global_CheckCast(value, "Libraries.Language.Types.Boolean");
    
    var me = this.valueOf();
    var other = b.GetValue();
    if ((me == other)) {
        return 0;
    }
    else if ((me < other)) {
        return -1;
    }
    else {
        return 1;
    }
};

File: Quorum/Library/Standard/Plugins/python/Libraries/Language/Types/Boolean.py
def Bool_(value):
	val = quorum_Libraries_Language_Types_Boolean_()
	val.SetValue__quorum_boolean(value)
	return val

class plugins_quorum_Libraries_Language_Types_Boolean_:
	def __init__(self, value = False):
		self.myBoolean = value

	def SetValueNative__quorum_boolean(self, bool):
		myBoolean = bool;
    
	def GetHashCode():
		if self.myBoolean:
			return 1231
		else:
			return 1237

